lib/internal/test_runner/runner.js

```diff
@@ -503,8 +506,8 @@ function run(options = kEmptyObject) {
   let filesWatcher;
   const opts = { __proto__: null, root, signal, inspectPort, testNamePatterns, only };
   if (watch) {
-    filesWatcher = watchFiles(testFiles, opts);
-    postRun = undefined;
+    filesWatcher = watchFiles(testFiles, { __proto__: null, ...opts, coverage });
```

Comment:
why not change `opts` directly?
```javascript
  const opts = { __proto__: null, root, signal, inspectPort, testNamePatterns, only, coverage };
  if (watch) {
    filesWatcher = watchFiles(testFiles, opts);
```

Reply:
that will make coverage run for every single test file, which I don't think we expect... right?

with this code output will be:

```bash

```